File: KutuphaneAPI/Entities/Models/Account.cs
using Microsoft.AspNetCore.Identity;

namespace Entities.Models
{
    public class Account : IdentityUser
    {
        public String? AvatarUrl { get; set; } = "avatars/default.png";
        public String? FirstName { get; set; }
        public String? LastName { get; set; }
        public DateTime MembershipDate { get; set; } = DateTime.UtcNow;
        public DateOnly? BirthDate { get; set; }
        public DateTime? LastLoginDate { get; set; }
        public String? RefreshToken { get; set; }
        public DateTime RefreshTokenExpiryTime { get; set; }
        public int CartId { get; set; }
        public Cart? Cart { get; set; }
        public ICollection<UserReview>? Reviews { get; set; }
        public ICollection<Reservation>? Reservations { get; set; }
        public ICollection<Loan>? Loans { get; set; }
        public ICollection<Penalty>? Penalties { get; set; }
        public ICollection<Notification>? Notifications { get; set; }
    }
}

File: KutuphaneWeb/library-automation-web/src/types/account.ts
export default interface Account {
    id?: string;
    userName?: string;
    email?: string;
    avatarUrl?: string;
    firstName?: string;
    lastName?: string;
    phoneNumber?: string;
    membershipDate?: Date;
    displayMembershipDate?: string;
    birthDate?: Date;
    displayBirthDate?: string;
    lastLoginDate?: Date;
    displayLastLoginDate?: string;
    roles?: string[];
}